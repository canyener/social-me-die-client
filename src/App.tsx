import React, { Component } from 'react'
import { Header, Icon, List } from 'semantic-ui-react'

import './App.css'
import axios from 'axios'

class App extends Component {
  state = {
    values: []
  }

  componentDidMount() {
    axios.get('http://localhost:5000/api/values').then(response => {
      this.setState(() => ({ values: response.data }))
    })
  }

  render() {
    return (
      <div>
        <Header as='h2'>
          <Icon name='users' />
          <Header.Content>Social Me Die</Header.Content>
        </Header>
        <List>
          {
            this.state.values.map((value: any) => 
              <List.Item key={value.id}>
                {value.name}
              </List.Item>)
          }
        </List>
      </div>
    )
  }
}

export default App
