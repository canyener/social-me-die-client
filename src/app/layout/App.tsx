import React, { useEffect, Fragment, useContext } from 'react'
import { Container } from 'semantic-ui-react'
import { observer } from 'mobx-react-lite'
import { Route, RouteComponentProps, withRouter } from 'react-router-dom'

import NavBar from '../../features/nav/NavBar'
import ActivityDashboard from '../../features/activities/dashboard/ActivityDashboard'
import LoadingComponent from './LoadingComponent'

import ActivityStore from '../stores/activityStore'
import HomePage from '../../features/home/HomePage'
import ActivityForm from '../../features/activities/form/ActivityForm'
import ActivityDetails from '../../features/activities/details/ActivityDetails'

const App: React.FC<RouteComponentProps> = ({ location }) => {

  const activityStore = useContext(ActivityStore)

  useEffect(() => {
    activityStore.loadActivities()
  }, [activityStore])

  if (activityStore.loadingInitial) return <LoadingComponent content="Loading Activities..." />

  return (
    <Fragment>
      <NavBar />
      <Container style={{ marginTop: '7em' }}>
        <Route exact path='/' component={HomePage} />
        <Route exact path='/activities' component={ActivityDashboard} />
        <Route path='/activities/:id' component={ActivityDetails} />
        <Route
          key={location.key}
          path={['/createActivity', '/manage/:id']}
          component={ActivityForm}
        />
      </Container>
    </Fragment>
  )
}

export default withRouter(observer(App))
