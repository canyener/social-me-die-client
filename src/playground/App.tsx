import React, { Component } from 'react'
import { Header, Icon, List } from 'semantic-ui-react'

import axios from 'axios'

import { IActivity } from '../app/models/activity'

interface IState {
  activities: IActivity[]
}

class App extends Component<{}, IState> {
  readonly state : IState = {
    activities: []
  }

  componentDidMount() {
    axios.get<IActivity[]>('http://localhost:5000/api/activities').then(response => {
      this.setState(() => ({ 
        activities: response.data 
      }))
    })
  }

  render() {
    return (
      <div>
        <Header as='h2'>
          <Icon name='users' />
          <Header.Content>Social Me Die</Header.Content>
        </Header>
        <List>
          {
            this.state.activities.map((activity) => 
              <List.Item key={activity.id}>
                {activity.title}
              </List.Item>)
          }
        </List>
      </div>
    )
  }
}

export default App
