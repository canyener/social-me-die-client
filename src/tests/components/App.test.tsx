import React from 'react'
import { shallow } from 'enzyme'

import App from '../../App'

const flushPromises = () => new Promise(resolve => setImmediate(resolve))

test('Should render App correctly', () => {
  const wrapper = shallow(<App />)
  expect(wrapper).toMatchSnapshot()
})

test('Should render App correctly with data', async () => {
  const wrapper = shallow(<App />)
  await flushPromises()
  wrapper.update()
  expect(wrapper).toMatchSnapshot()
})
