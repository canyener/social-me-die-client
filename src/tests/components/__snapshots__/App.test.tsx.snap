// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render App correctly when activities are fetched successfully 1`] = `
<App>
  <Component>
    <Menu
      fixed="top"
      inverted={true}
    >
      <div
        className="ui inverted top fixed menu"
      >
        <Container>
          <div
            className="ui container"
          >
            <MenuItem
              header={true}
            >
              <div
                className="header item"
                onClick={[Function]}
              >
                <img
                  alt="logo"
                  src="/assets/logo.png"
                  style={
                    Object {
                      "marginRight": "10px",
                    }
                  }
                />
                Social Me Die
              </div>
            </MenuItem>
            <MenuItem
              name="Activities"
            >
              <div
                className="item"
                onClick={[Function]}
              >
                Activities
              </div>
            </MenuItem>
            <MenuItem>
              <div
                className="item"
                onClick={[Function]}
              >
                <Button
                  as="button"
                  content="Create Activity"
                  positive={true}
                >
                  <Ref
                    innerRef={
                      Object {
                        "current": <button
                          class="ui positive button"
                        >
                          Create Activity
                        </button>,
                      }
                    }
                  >
                    <RefFindNode
                      innerRef={
                        Object {
                          "current": <button
                            class="ui positive button"
                          >
                            Create Activity
                          </button>,
                        }
                      }
                    >
                      <button
                        className="ui positive button"
                        onClick={[Function]}
                      >
                        Create Activity
                      </button>
                    </RefFindNode>
                  </Ref>
                </Button>
              </div>
            </MenuItem>
          </div>
        </Container>
      </div>
    </Menu>
  </Component>
  <Container
    style={
      Object {
        "marginTop": "7em",
      }
    }
  >
    <div
      className="ui container"
      style={
        Object {
          "marginTop": "7em",
        }
      }
    >
      <Component
        activities={
          Array [
            Object {
              "category": "Fake Category 1",
              "city": "Fake City",
              "date": 1970-01-01T00:00:01.000Z,
              "description": "Fake description",
              "id": "1",
              "title": "Fake Activity 1",
              "venue": "Fake Venue",
            },
            Object {
              "category": "Fake Category 2",
              "city": "Fake City 2",
              "date": 1970-01-01T00:00:01.000Z,
              "description": "Fake description 2",
              "id": "2",
              "title": "Fake Activity 2",
              "venue": "Fake Venue 2",
            },
          ]
        }
      >
        <Grid>
          <div
            className="ui grid"
          >
            <GridColumn
              width={10}
            >
              <div
                className="ten wide column"
              >
                <Component
                  activities={
                    Array [
                      Object {
                        "category": "Fake Category 1",
                        "city": "Fake City",
                        "date": 1970-01-01T00:00:01.000Z,
                        "description": "Fake description",
                        "id": "1",
                        "title": "Fake Activity 1",
                        "venue": "Fake Venue",
                      },
                      Object {
                        "category": "Fake Category 2",
                        "city": "Fake City 2",
                        "date": 1970-01-01T00:00:01.000Z,
                        "description": "Fake description 2",
                        "id": "2",
                        "title": "Fake Activity 2",
                        "venue": "Fake Venue 2",
                      },
                    ]
                  }
                >
                  <Segment
                    clearing={true}
                  >
                    <div
                      className="ui clearing segment"
                    >
                      <ItemGroup
                        divided={true}
                      >
                        <div
                          className="ui divided items"
                        >
                          <Item
                            key="1"
                          >
                            <div
                              className="item"
                            >
                              <ItemContent>
                                <div
                                  className="content"
                                >
                                  <ItemHeader
                                    as="a"
                                  >
                                    <a
                                      className="header"
                                    >
                                      Fake Activity 1
                                    </a>
                                  </ItemHeader>
                                  <ItemMeta>
                                    <div
                                      className="meta"
                                    >
                                      01.01.1970 00:00:01
                                    </div>
                                  </ItemMeta>
                                  <ItemDescription>
                                    <div
                                      className="description"
                                    >
                                      <div>
                                        Fake description
                                      </div>
                                      <div>
                                        Fake City
                                        , 
                                        Fake Venue
                                      </div>
                                    </div>
                                  </ItemDescription>
                                  <ItemExtra>
                                    <div
                                      className="extra"
                                    >
                                      <Button
                                        as="button"
                                        color="blue"
                                        content="View"
                                        floated="right"
                                      >
                                        <Ref
                                          innerRef={
                                            Object {
                                              "current": <button
                                                class="ui blue right floated button"
                                              >
                                                View
                                              </button>,
                                            }
                                          }
                                        >
                                          <RefFindNode
                                            innerRef={
                                              Object {
                                                "current": <button
                                                  class="ui blue right floated button"
                                                >
                                                  View
                                                </button>,
                                              }
                                            }
                                          >
                                            <button
                                              className="ui blue right floated button"
                                              onClick={[Function]}
                                            >
                                              View
                                            </button>
                                          </RefFindNode>
                                        </Ref>
                                      </Button>
                                      <Label
                                        basic={true}
                                        content="Fake Category 1"
                                      >
                                        <div
                                          className="ui basic label"
                                          onClick={[Function]}
                                        >
                                          Fake Category 1
                                        </div>
                                      </Label>
                                    </div>
                                  </ItemExtra>
                                </div>
                              </ItemContent>
                            </div>
                          </Item>
                          <Item
                            key="2"
                          >
                            <div
                              className="item"
                            >
                              <ItemContent>
                                <div
                                  className="content"
                                >
                                  <ItemHeader
                                    as="a"
                                  >
                                    <a
                                      className="header"
                                    >
                                      Fake Activity 2
                                    </a>
                                  </ItemHeader>
                                  <ItemMeta>
                                    <div
                                      className="meta"
                                    >
                                      01.01.1970 00:00:01
                                    </div>
                                  </ItemMeta>
                                  <ItemDescription>
                                    <div
                                      className="description"
                                    >
                                      <div>
                                        Fake description 2
                                      </div>
                                      <div>
                                        Fake City 2
                                        , 
                                        Fake Venue 2
                                      </div>
                                    </div>
                                  </ItemDescription>
                                  <ItemExtra>
                                    <div
                                      className="extra"
                                    >
                                      <Button
                                        as="button"
                                        color="blue"
                                        content="View"
                                        floated="right"
                                      >
                                        <Ref
                                          innerRef={
                                            Object {
                                              "current": <button
                                                class="ui blue right floated button"
                                              >
                                                View
                                              </button>,
                                            }
                                          }
                                        >
                                          <RefFindNode
                                            innerRef={
                                              Object {
                                                "current": <button
                                                  class="ui blue right floated button"
                                                >
                                                  View
                                                </button>,
                                              }
                                            }
                                          >
                                            <button
                                              className="ui blue right floated button"
                                              onClick={[Function]}
                                            >
                                              View
                                            </button>
                                          </RefFindNode>
                                        </Ref>
                                      </Button>
                                      <Label
                                        basic={true}
                                        content="Fake Category 2"
                                      >
                                        <div
                                          className="ui basic label"
                                          onClick={[Function]}
                                        >
                                          Fake Category 2
                                        </div>
                                      </Label>
                                    </div>
                                  </ItemExtra>
                                </div>
                              </ItemContent>
                            </div>
                          </Item>
                        </div>
                      </ItemGroup>
                    </div>
                  </Segment>
                </Component>
              </div>
            </GridColumn>
          </div>
        </Grid>
      </Component>
    </div>
  </Container>
</App>
`;

exports[`Should render App correctly with empty activity list 1`] = `
<App>
  <Component>
    <Menu
      fixed="top"
      inverted={true}
    >
      <div
        className="ui inverted top fixed menu"
      >
        <Container>
          <div
            className="ui container"
          >
            <MenuItem
              header={true}
            >
              <div
                className="header item"
                onClick={[Function]}
              >
                <img
                  alt="logo"
                  src="/assets/logo.png"
                  style={
                    Object {
                      "marginRight": "10px",
                    }
                  }
                />
                Social Me Die
              </div>
            </MenuItem>
            <MenuItem
              name="Activities"
            >
              <div
                className="item"
                onClick={[Function]}
              >
                Activities
              </div>
            </MenuItem>
            <MenuItem>
              <div
                className="item"
                onClick={[Function]}
              >
                <Button
                  as="button"
                  content="Create Activity"
                  positive={true}
                >
                  <Ref
                    innerRef={
                      Object {
                        "current": <button
                          class="ui positive button"
                        >
                          Create Activity
                        </button>,
                      }
                    }
                  >
                    <RefFindNode
                      innerRef={
                        Object {
                          "current": <button
                            class="ui positive button"
                          >
                            Create Activity
                          </button>,
                        }
                      }
                    >
                      <button
                        className="ui positive button"
                        onClick={[Function]}
                      >
                        Create Activity
                      </button>
                    </RefFindNode>
                  </Ref>
                </Button>
              </div>
            </MenuItem>
          </div>
        </Container>
      </div>
    </Menu>
  </Component>
  <Container
    style={
      Object {
        "marginTop": "7em",
      }
    }
  >
    <div
      className="ui container"
      style={
        Object {
          "marginTop": "7em",
        }
      }
    >
      <Component
        activities={Array []}
      >
        <Grid>
          <div
            className="ui grid"
          >
            <GridColumn
              width={10}
            >
              <div
                className="ten wide column"
              >
                <Component
                  activities={Array []}
                >
                  <Segment
                    clearing={true}
                  >
                    <div
                      className="ui clearing segment"
                    >
                      <ItemGroup
                        divided={true}
                      >
                        <div
                          className="ui divided items"
                        />
                      </ItemGroup>
                    </div>
                  </Segment>
                </Component>
              </div>
            </GridColumn>
          </div>
        </Grid>
      </Component>
    </div>
  </Container>
</App>
`;
